### ISSUE TITLE
generateProto gradle task error: %1 is not a valid Win32 application.

### ISSUE THREAD

**Issue by vasiliypelikh** [akka/akka-grpc#602]
Java version: 11
Gradle version: 5.4.1
Akka-grpc plugin version: 0.6.2

During building I see error:
```
Execution failed for task ':commons:generateProto'.
> protoc: stdout: . stderr: P:\pvp-game\commons\src\main\protobuf: warning: directory does not exist.
  P:\pvp-game\commons\app\protobuf: warning: directory does not exist.
  P:\pvp-game\commons\app\proto: warning: directory does not exist.
  --akkaGrpc_out: protoc-gen-akkaGrpc: %1 is not a valid Win32 application.
```

Full gradle log with debug: [gradlelog.txt](https://github.com/akka/akka-grpc/files/3290043/gradlelog.txt)

**Comment by johanandren** [comment]
Not quite the same but #576 may perhaps provide some hints why/how the protoc command isn't extracted/runnable on windows as well.

**Comment by vasiliypelikh** [comment]
I tried to set java.io.tmpdir during building (gradlew -Djava.io.tmpdir=P:\qqw\commons\build\proto :commons:generateProto). Still same error.
Also If I use protoc executable directly it generate java files.

**Comment by raboof** [comment]
The akka-grpc gradle plugin uses calls the gradle protoc plugin with the akka-grpc codegen jars as parameters (`com.lightbend.akka.grpc:akka-grpc-codegen_2.12:${pluginVersion}:assembly@jar` and, if targeting scala, also `com.lightbend.akka.grpc:akka-grpc-scalapb-protoc-plugin_2.12:${pluginVersion}:assembly@jar`.

If I recall correctly the `@jar` tells protoc to treat this generator as a jar rather than as a native executable. I suspect the `%1` from the error message was intended to be replaced by the artifact. It's unfortunate that that doesn't work, but I would bet it is one of these 2 jars.

Perhaps protoc does not support treating generators as jars instead of native executables on Windows, and here reports it fails to run the jar as a native executable?

**Comment by mlangc** [comment]
I've hacked together a workaround that could be evolved into a proper fix here: https://github.com/mlangc/akka-grpc/commit/bc37a2222853385a8e9e999151d549c7caad60ad

The changes I've made are:
*  Prepending the "default universial script" instead of the "default shell script": https://github.com/mlangc/akka-grpc/commit/bc37a2222853385a8e9e999151d549c7caad60ad#diff-fdc3abdfd754eeb24090dbd90aeec2ceL26
* Rename the plugin assembly jars into `bat` before passing them to the plugin: https://github.com/mlangc/akka-grpc/commit/bc37a2222853385a8e9e999151d549c7caad60ad#diff-ee344aab74dcfb1a70eb10f7d1f113bbR29 This is at this stage quite hacky and should propably be reworked. Suggestions are welcome - I'm quite new to gradle!

**Comment by raboof** [comment]
Thanks for sharing, that indeed looks promising!

**Comment by mlangc** [comment]
Does it make sense to clean this up and create a PR (where further details can be sorted out before merging)?

**Comment by raboof** [comment]
Yes, that would definitely be welcome! I see there's several independent improvements in there, perhaps those could even be separate PR's so the uncontroversial ones can be merged quickly.

**Comment by mlangc** [comment]
Alright, I'll look into it in the next few days!

**Comment by raboof** [comment]
Would be great! Much appreciated!

**Comment by mlangc** [comment]
I've just created #602, hoping that with your feedback I can evolve into something that you can merge.

**Comment by mlangc** [comment]
@raboof it seems that this issue has been incorrectly closed by merging #621

**Comment by raboof** [comment]
Oops, didn't notice that!
